tests/test_profiling_result_printer.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) 2025, Han-Xuan Huang <c1ydehhx@gmail.com>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# - Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+# - Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+# - Neither the name of the copyright holder nor the names of its contributors
+#   may be used to endorse or promote products derived from this software
+#   without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+from typing import Any
+
+import pytest
+import numpy
+
+import modmesh
+from modmesh.profiling import (
+    ProfilingResultPrinter,
+)
+
+
+class TestProfilingResultPrinter:
+    def run_profile_function(self):
+        _ = modmesh.CallProfilerProbe("numpy_arange_100")
+        numpy.arange(0, 100, dtype="uint32")
+
+    @pytest.fixture
+    def profiling_result_fixture(self) -> dict[str, Any]:
+        modmesh.call_profiler.reset()
+
+        self.run_profile_function()
+
+        result: dict[str, Any] = modmesh.call_profiler.result()["children"]
+        return result
+
+    def test_getitem_valid_key(
+        self, profiling_result_fixture: dict[str, Any]
+    ) -> None:
+        printer: ProfilingResultPrinter = ProfilingResultPrinter(
+            profiling_result_fixture
+        )
+
+        assert printer["numpy_arange_100"].name == "numpy_arange_100"
+
+    def test_getitem_absent_key(
+        self, profiling_result_fixture: dict[str, Any]
+    ) -> None:
+        printer: ProfilingResultPrinter = ProfilingResultPrinter(
+            profiling_result_fixture
+        )
+
+        try:
+            printer["numpy_arange_1000"]
+            assert False
```

Comment:
Never use `assert` in modmesh testing code.

Use helper functions in https://docs.python.org/3/library/unittest.html#unittest.TestCase .  Using [`assertRaisesRegex`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex), you can [test both the exception and error messages](https://github.com/solvcon/modmesh/blob/931e3a3bf7949da9d3e2f4952307e9ae1dae535b/tests/test_buffer.py#L51).  Your testing code here can be improved like:

```python
        with self.assertRaisesRegex(
                ValueError,
                "... (please find it out yourself) ..."
        ):
            printer["numpy_arange_1000"]
```